docs/source/contributing/contributing-code.rst

```diff
@@ -0,0 +1,90 @@
+Contributing code
+==================================================
+
+Local setup
+----------------------------------
+
+To contribute code, basic knowledge of git is required, if you are unfamiliar
+with the git workflow, checkout `GitHub's tutorial <https://docs.github.com/en/get-started/quickstart/hello-world>`_.
+The following steps allow you to contribute code to DoWhy.
+
+1. Fork the `dowhy main repository <https://github.com/py-why/dowhy>`_
+2. Clone this repository to your local machine using
+
+.. code:: shell
+
+   git clone https://github.com/<YOUR_GITHUB_USERNAME>/dowhy
+
+3. [optional] Create a virtual environment using python venv or conda #check if poetry does this automatically
+
+.. code:: shell
+
+   virtualenv dowhy-env --python=python3.8
+   source dowhy-env/bin/activate
+
+Note this step is not necessary as Poetry will create a new environment by itself if needed.
+
+4. Install DoWhy and it's requirements in your environment.
+By default, Poetry will install DoWhy in interactive mode.
+This way, you can immediately test your changes to the codebase.
+
+.. code:: shell
+
+   cd dowhy
+   pip install --upgrade pip
+   poetry install -E "plotting causalml docs"
+
+.. note::
+   Installing pygraphviz can cause problems on some platforms.
+   Please consult the documentation of `pygraphviz <https://pygraphviz.github.io/documentation/stable/install.html>`_
+
+5. (optional) add dowhy as an upstream remote to keep your
+fork up-to-date with DoWhy's main branch.
+
+.. code:: shell
+
+   git remote add upstream http://www.github.com/py-why/dowhy
+
+You are now ready to make changes to the code base locally.
+
+Pull request checklist
+----------------------------------
+
+1. Execute the flake8 linter for breaking warnings and fix all reported problems.
+
+.. code:: shell
+
+  poetry run poe lint
+
+2. Add tests for your new code and execute the unittests to make sure
+you did not introduce any breaking changes or bugs.
+
+.. code:: shell
+
+  poetry run poe test
+
+Note that you can also execute those tasks together
+
+.. code:: shell
+
+  poetry run poe verify
+A full list of available tasks can be obtained using
+
+.. code:: shell
+
+  poetry run poe -h
+
+The full test suite of DoWhy takes quite long. To speed up development cycles,
+you can restrict the tests executed as in the following example.
+
+.. code:: shell
+
+  pytest -v tests/causal_refuters
```

Comment:
Because `pytest` is in poetry`s venv, I'd change this to: 

`poetry run py`test -v tests/causal_refuters`

Reply:
totally, thanks